README.md
# Recent Global Posts

**INACTIVE NOTICE: This plugin is unsupported by WPMUDEV, we've published it here for those technical types who might want to fork and maintain it for their needs.**

## Translations

Translation files can be found at https://github.com/wpmudev/translations

## Recent Global Posts allows you to display a list of recent posts from across your Multisite network on your main site.

### Built for your network

This plugin uses the power of [Post Indexer](http://premium.wpmudev.org/project/post-indexer/ "Post Indexer") to publish recent posts anywhere on your main site dynamically using shortcodes – on pages, posts and widget areas, such as a sidebar or footer. Built and supported by WordPress network experts, Recent Network Posts provides the safe reliable service you need for operating a successful Multisite or BuddyPress community. 

![image](http://premium.wpmudev.org/wp-content/uploads/2009/06/latestposts.jpg)

 Help users discover new content from across your network and strengthen your sense of community with Recent Global Posts.

## Usage

### To Get Started:

Start by reading the [Installing Plugins](https://premium.wpmudev.org/wpmu-manual/installing-regular-plugins-on-wpmu/) section in our comprehensive [WordPress and WordPress Multisite Manual](https://premium.wpmudev.org/wpmu-manual/) if you are new to WordPress. _This plugin requires the [Post Indexer](https://premium.wpmudev.org/project/post-indexer) plugin_ _If you have an older version of Network Recent Posts installed in /mu-plugins/ please delete it._ Once installed and _network-activated_, it's ready to use. There's no configuration necessary!

### Displaying your network posts

You can display your recent posts anywhere on your site simply by adding a shortcode. The base shortcode is:

    [globalrecentposts]

That will display a simple list of linked titles to the 5 most recent posts in your network. 

![Recent Network Posts Basic Display](https://premium.wpmudev.org/wp-content/uploads/2009/06/recent-network-posts-3000-basic-list.png)

 But you can use the following attributes to really customize what content to display, and how to display it:

*   _number="5"_ - How many posts you want to display
*   _title_characters="250"_ - Maximum number of characters in each title.
*   _content_characters="200"_ - Maximum number of characters in the content of each entry
*   _title_content_divider="-"_ - What to use to separate the title from the content. If this parameter is not included, the content will display beneath the title.
*   _title_link="no"_ - By default, the title links to the post. You can use this to remove the link.
*   _show_avatars="yes"_ - Displays the author avatar if avatars are used on your site.
*   _avatar_size="32"_ - Sets the square size of the avatars.
*   _posttype="post-type"_ - Use to specify the post type to display. Default is "post". Note that you can only specify one post-type.

So, for example, if you want to show the 3 most recent posts with a little excerpt and author avatar, you could use it like this:

    [globalrecentposts number="3" content_characters="200" show_avatars="yes" avatar_size="32"]

The above shortcode would produce something like this on your site: 

![Recent Network Posts Avatar List](https://premium.wpmudev.org/wp-content/uploads/2009/06/recent-network-posts-3000-avatar-list.png)

 Don't like the default styling? You can also use these parameters to add text or HTML elements to really customize the content, as well as layout & styling:

*   _title_before="text or HTML element"_ - Executes before the title of each entry.
*   _title_after="text or HTML element"_ - Executes after the title of each entry.
*   _global_before="text or HTML element"_ - Executes above the list of entries.
*   _global_after="text or HTML element"_ - Executes beneath the list of entries.
*   _before="text or HTML element"_ - Executes before each entry.
*   _after="text or HTML element"_ - Executes after each entry.

Here's an example shortcode with some of these extra parameters. You'll notice we've included headline tags for the titles, and a custom class for each entry so we can style the output with a bit of custom CSS.

[globalrecentposts number="2" content_characters="200" global_before="Here are our most recent posts. Enjoy!" before="" after="" show_avatars="yes" avatar_size="48" title_before="" title_after=""]

The above shortcode could display something like this (note that the actual CSS to be used would depend a lot on your theme, so we haven't included that here). 

![Recent Network Posts Styled List](https://premium.wpmudev.org/wp-content/uploads/2009/06/recent-network-posts-3000-styled-list.png)


changelog.txt
Plugin Name: Recent Posts
Author: Andrew Billits

Change Log:
----------------------------------------------------------------------
----------------------------------------------------------------------
2.1 - 05/11/2011
----------------------------------------------------------------------
- Added post type function parameter
- Added shortcode

2.0 - 01/08/2011
----------------------------------------------------------------------
- Update for version 3.1

1.0.1 - 08/15/2009
----------------------------------------------------------------------
- Added avatar support.

1.0.0 - 01/15/2009
----------------------------------------------------------------------
- Initial Release.

recent-posts.php
<?php
/*
Plugin Name: Recent Posts
Plugin URI:
Description: Recent posts function and shortcode
Author: Andrew Billits (Incsub)
Version: 2.1
Author URI:
WDP ID: 75
*/

/*
Copyright 2007-2009 Incsub (http://incsub.com)

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License (Version 2 - GPLv2) as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

/*
Usage:
display_recent_posts(NUMBER,TITLE_CHARACTERS,CONTENT_CHARACTERS,TITLE_CONTENT_DIVIDER,TITLE_BEFORE,TITLE_AFTER,GLOBAL_BEFORE,GLOBAL_AFTER,BEFORE,AFTER,TITLE_LINK,SHOW_AVATARS,AVATAR_SIZE,POSTTYPE, ECHO);

Ex:
display_recent_posts(10,40,150,'<br />','<strong>','</strong>','<ul>','</ul>','<li>','</li>','yes','yes',16, 'post', true);
*/
/* -------------------- Update Notifications Notice -------------------- */
if ( !function_exists( 'wdp_un_check' ) ) {
  add_action( 'admin_notices', 'wdp_un_check', 5 );
  add_action( 'network_admin_notices', 'wdp_un_check', 5 );
  function wdp_un_check() {
    if ( !class_exists( 'WPMUDEV_Update_Notifications' ) && current_user_can( 'edit_users' ) )
      echo '<div class="error fade"><p>' . __('Please install the latest version of <a href="http://premium.wpmudev.org/project/update-notifications/" title="Download Now &raquo;">our free Update Notifications plugin</a> which helps you stay up-to-date with the most stable, secure versions of WPMU DEV themes and plugins. <a href="http://premium.wpmudev.org/wpmu-dev/update-notifications-plugin-information/">More information &raquo;</a>', 'wpmudev') . '</a></p></div>';
  }
}
/* --------------------------------------------------------------------- */

//------------------------------------------------------------------------//
//---Config---------------------------------------------------------------//
//------------------------------------------------------------------------//

//------------------------------------------------------------------------//
//---Hook-----------------------------------------------------------------//
//------------------------------------------------------------------------//

//------------------------------------------------------------------------//
//---Functions------------------------------------------------------------//
//------------------------------------------------------------------------//

//------------------------------------------------------------------------//
//---Output Functions-----------------------------------------------------//
//------------------------------------------------------------------------//

function display_recent_posts($tmp_number,$tmp_title_characters = 0,$tmp_content_characters = 0,$tmp_title_content_divider = '<br />',$tmp_title_before,$tmp_title_after,$tmp_global_before,$tmp_global_after,$tmp_before,$tmp_after,$tmp_title_link = 'no',$tmp_show_avatars = 'yes', $tmp_avatar_size = 16, $posttype = 'post', $output = true){
	global $wpdb;
	$query = $wpdb->prepare("SELECT * FROM " . $wpdb->base_prefix . "site_posts WHERE blog_public = '1' AND post_type = %s ORDER BY post_published_stamp DESC LIMIT %d", $posttype, $tmp_number);
	$tmp_posts = $wpdb->get_results( $query, ARRAY_A );

	$html = '';

	if (count($tmp_posts) > 0){
		$html .= $tmp_global_before;

		$default_avatar = get_option('default_avatar');
		foreach ($tmp_posts as $tmp_post){

			$html .= $tmp_before;
			if ( $tmp_title_characters > 0 ) {
				$html .= $tmp_title_before;
				if ( $tmp_show_avatars == 'yes' ) {
					if ( $tmp_title_link == 'no' ) {
						$html .= get_avatar( $tmp_post['post_author'], $tmp_avatar_size, $default_avatar) . ' ' . substr($tmp_post['post_title'],0,$tmp_title_characters);
					} else {
						$html .= get_avatar( $tmp_post['post_author'], $tmp_avatar_size, $default_avatar) . ' <a href="' . $tmp_post['post_permalink'] . '" style="text-decoration:none;">' . substr($tmp_post['post_title'],0,$tmp_title_characters) . '</a>';
					}
				} else {
					if ( $tmp_title_link == 'no' ) {
						$html .= substr($tmp_post['post_title'],0,$tmp_title_characters);
					} else {
						$html .= '<a href="' . $tmp_post['post_permalink'] . '" style="text-decoration:none;">' . substr($tmp_post['post_title'],0,$tmp_title_characters) . '</a>';
					}
				}
				$html .= $tmp_title_after;
			}
			$html .= $tmp_title_content_divider;
			if ( $tmp_content_characters > 0 ) {
				$html .= substr($tmp_post['post_content_stripped'],0,$tmp_content_characters);
			}
			$html .= $tmp_after;
		}
		$html .= $tmp_global_after;

	}

	if($output) {
		echo $html;
	} else {
		return $html;
	}

}

function display_recent_posts_shortcode($atts, $content = null, $code = "") {

	$defaults = array(	'number'	=>	5,
						'title_characters' => 250,
						'content_characters' => 0,
						'title_content_divider' => '<br />',
						'title_before'	=>	'',
						'title_after'	=>	'',
						'global_before'	=>	'<ul>',
						'global_after'	=>	'</ul>',
						'before'	=>	'<li>',
						'after'	=>	'</li>',
						'title_link' => 'yes',
						'show_avatars' => 'no',
						'avatar_size' => 16,
						'posttype' => 'post'
					);

	extract(shortcode_atts($defaults, $atts));

	$html = '';

	$html .= display_recent_posts( $number, $title_characters, $content_characters, $title_content_divider, $title_before, $title_after, $global_before, $global_after, $before, $after, $title_link, $show_avatars, $avatar_size, $posttype, false);

	return $html;

}
add_shortcode( 'globalrecentposts', 'display_recent_posts_shortcode');


//------------------------------------------------------------------------//
//---Page Output Functions------------------------------------------------//
//------------------------------------------------------------------------//

//------------------------------------------------------------------------//
//---Support Functions----------------------------------------------------//
//------------------------------------------------------------------------//

?>